integration/mediation-tests/tests-mediator-1/src/test/java/org/wso2/carbon/esb/mediator/test/cache/ESBJAVA4721PIWithCacheTestCase.java

```diff
@@ -0,0 +1,78 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.wso2.carbon.esb.mediator.test.cache;
+
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.carbon.automation.engine.exceptions.AutomationFrameworkException;
+import org.wso2.carbon.automation.test.utils.http.client.HttpRequestUtil;
+import org.wso2.carbon.automation.test.utils.http.client.HttpResponse;
+import org.wso2.esb.integration.common.utils.ESBIntegrationTest;
+import org.wso2.esb.integration.common.utils.common.ServerConfigurationManager;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * This test the ability of JSON to XML conversion with processing instructions enabled when caching is enabled.
+ */
+public class ESBJAVA4721PIWithCacheTestCase extends ESBIntegrationTest {
+
+    private ServerConfigurationManager serverConfigurationManager;
+
+    @BeforeClass(alwaysRun = true)
+    public void deployArtifacts() throws Exception {
+        super.init();
+        serverConfigurationManager = new ServerConfigurationManager(context);
+        serverConfigurationManager.applyConfiguration(new File(
+                getESBResourceLocation() + File.separator + "mediatorconfig" + File.separator + "cache" + File.separator
+                        + "synapse.properties"));
+        super.init();
+        loadESBConfigurationFromClasspath(
+                File.separator + "artifacts" + File.separator + "ESB" + File.separator + "mediatorconfig"
+                        + File.separator + "cache" + File.separator + "CacheMediatorforJasonResponse.xml");
+    }
+
+    @Test(groups = "wso2.esb",
+          description = "Test cache mediator with  Json response having a single element array with PI enabled")
+    public void testJsonResponseWithCacheMediator() throws IOException, AutomationFrameworkException {
+
+        Map<String, String> requestHeader = new HashMap<>();
+        requestHeader.put("Content-type", "application/json");
+
+        //will not be a cache hit
+        HttpRequestUtil.doGet((getApiInvocationURL("jsonTest") + "/backend"), requestHeader);
+
+        //will be a cache hit
+        HttpResponse response = HttpRequestUtil.
+                doGet((getApiInvocationURL("jsonTest") + "/backend"), requestHeader);
+
+        //check if [] are preserved in response
+        Assert.assertTrue(response.getData().contains("[\"water\"]"),
+                "Expected response was not" + " received. Got " + response.getData());
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void stop() throws Exception {
+        super.cleanup();
```

Comment:
Don't we need to restore the server configurations to initial state to avoid effecting this change in synapse.properties file to other test cases